tests/sentry/uptime/detectors/test_url_extraction.py

```diff
@@ -0,0 +1,19 @@
+from sentry.testutils.cases import TestCase
+from sentry.uptime.detectors.url_extraction import extract_base_url
+
+
+class ExtractBaseUrlTest(TestCase):
+    def run_test(self, url: str, expected_url: str | None):
+        assert extract_base_url(url) == expected_url
+
+    def test(self):
+        self.run_test("", None)
+        self.run_test("192.168.0.1", None)
+        self.run_test("https://192.168.0.1", None)
+        self.run_test("https://www.sentry.io/", "https://www.sentry.io")
+        self.run_test("https://www.sentry.io/some/page", "https://www.sentry.io")
+        self.run_test("www.sentry.io", None)
+        self.run_test("www.sentry.io/something", None)
+        self.run_test("https://www.sentry.nope/", None)
+        self.run_test("https://www.sentry.nope/some/page", None)
+        self.run_test("https://www.ato.gov.au/", "https://www.ato.gov.au")
```

Comment:
Probably could drop the class based test and use the pytest parameterize thing like here https://github.com/getsentry/sentry/blob/495754e103f72dab80b8474640f38e6f75c12de6/tests/sentry/security/test_csp.py#L6-L20